Source/WebCore/Modules/permissions/Permissions.cpp

```diff
@@ -77,66 +82,77 @@ static bool isAllowedByFeaturePolicy(const Document& document, PermissionName na
 
 void Permissions::query(JSC::Strong<JSC::JSObject> permissionDescriptorValue, DOMPromiseDeferred<IDLInterface<PermissionStatus>>&& promise)
 {
-    // FIXME: support permissions in WorkerNavigator.
+    auto* context = m_navigator ? m_navigator->scriptExecutionContext() : nullptr;
+    auto* document = dynamicDowncast<Document>(*context);
```

Comment:
Good point, I could swear an earlier iteration has the null check... :/

Reply:
Oh wait, maybe I have misinterpreted something. I thought we were using `dynamicDowncast<Document>(*context);` to avoid a null check--as in, if context is null, then document will be null (and workerGlobalScope will be null when we dynamicDowncast cast it), and since those are being checked, we'll catch the issue if context it null. So then we don't need to check if context is null explicitly. Is that not the extra null check that we were trying to remove from the previous iteration? If not, what was the unneeded null check we were trying to remove?

Reply:
`dynamicDowncast<Document>(*context);` avoids a null check when we know `context` is not null. If context can be null then doing `*context` is wrong (you may be dereferencing null).

Reply:
The previous iteration had a previous null check for `context` with an early return. So I suggested using `dynamicDowncast<Document>(*context);` to avoid a second unnecessary null check of the context. However, in this iteration you dropped the context (and context->globalObject()) without explanation.

Reply:
Ah that makes sense. My apologies, I misunderstood. Fixed!